File: backend/app/schemas/a2i.py
from pydantic import BaseModel


class A2ITaskSummary(BaseModel):
    id: str
    documentId: str
    pageNumber: int
    humanLoopName: str | None
    status: str
    triggerReason: str
    reviewerId: str | None
    reviewTimestamp: str | None
    correctionApplied: bool
    confidenceScore: float | None
    s3OutputUri: str | None
    assignedTo: str | None
    assignedAt: str | None
    createdAt: str


class DiffItemSchema(BaseModel):
    id: str
    diffType: str        # changed_word | missing_word | extra_word | table_mismatch
    nativeValue: str
    textractValue: str
    lineIndex: int


class A2ITaskDetail(A2ITaskSummary):
    """Full task detail including diff items and text snapshots for the review UI."""
    diffItems: list[DiffItemSchema]
    nativeTextSnapshot: str | None
    originalTextractText: str | None


class A2ICompleteBody(BaseModel):
    correctedText: str
    reviewerId: str
    comment: str | None = None


class AssignTaskBody(BaseModel):
    reviewerId: str


class ReviewerStatsResponse(BaseModel):
    reviewerId: str
    totalAssigned: int
    completed: int
    pending: int
    correctionsApplied: int
    acceptanceRate: float   # percentage 0-100

File: frontend/src/types/a2i.ts
export type A2ITaskStatus =
  | 'pending'
  | 'assigned'
  | 'in_review'
  | 'under_review'
  | 'completed'
  | 'auto_verified'
  | 'failed';

export type DiffAction = 'accepted_textract' | 'accepted_native' | 'edited' | 'rejected' | null;

export interface DiffItem {
  id: string;
  diffType: 'changed_word' | 'missing_word' | 'extra_word' | 'table_mismatch';
  nativeValue: string;
  textractValue: string;
  lineIndex: number;
  action?: DiffAction;
  correctedValue?: string;
}

export interface A2ITask {
  id: string;
  documentId: string;
  pageNumber: number;
  humanLoopName?: string;
  status: A2ITaskStatus;
  triggerReason: string;
  reviewerId?: string;
  reviewTimestamp?: string;
  correctionApplied: boolean;
  confidenceScore?: number;
  s3OutputUri?: string;
  assignedTo?: string;
  assignedAt?: string;
  createdAt: string;
}

export interface A2ITaskDetail extends A2ITask {
  diffItems: DiffItem[];
  nativeTextSnapshot?: string;
  originalTextractText?: string;
}

export interface ReviewerStats {
  reviewerId: string;
  totalAssigned: number;
  completed: number;
  pending: number;
  correctionsApplied: number;
  acceptanceRate: number;
}
